test/classes/test_simplicial_complex.py

```diff
@@ -56,6 +56,98 @@ def test_is_maximal(self):
         is_maximal = sc.is_maximal([1, 2, 3])
         self.assertTrue(is_maximal)
 
+    def test_contructor_using_graph(self):
+        """Test input a networkx graph in the constructor."""
+        G = nx.Graph()
+        G.add_edge(0, 1)
+        G.add_edge(2, 5)
+        G.add_edge(5, 4, weight=5)
+        SC = SimplicialComplex(G)
+
+        assert (0, 1) in SC.simplices
+        assert (2, 5) in SC.simplices
+        assert (5, 4) in SC.simplices
+        assert SC.simplices[(5, 4)]["weight"] == 5
+
+    def test_skeleton_raise_errors(self):
+        """Test skeleton raises."""
+        with self.assertRaises(ValueError):
+            G = nx.Graph()
+            G.add_edge(0, 1)
+            G.add_edge(2, 5)
+            G.add_edge(5, 4, weight=5)
+            SC = SimplicialComplex(G)
+            SC.skeleton(-2)
+
+        with self.assertRaises(ValueError):
+            G = nx.Graph()
+            G.add_edge(0, 1)
+            G.add_edge(2, 5)
+            G.add_edge(5, 4, weight=5)
+            SC = SimplicialComplex(G)
+            SC.skeleton(2)
+
+    def test_rep_str(self):
```

Comment:
rename rep --> repr:

Because:
whenever possible, the name of the test function should be `test_{name of the function}` and here we are testing `__repr__`